src/megachatapi_impl.cpp

```diff
@@ -5287,7 +5287,7 @@ MegaChatRoomPrivate::MegaChatRoomPrivate(const ChatRoom &chat)
     this->chatid = chat.chatid();
     this->priv = (privilege_t) chat.ownPriv();
     this->group = chat.isGroup();
-    this->mPublicChat = ((GroupChatRoom &)chat).publicChat();
+    this->mPublicChat = this->group ?((GroupChatRoom &)chat).publicChat() :false;
```

Comment:
Better to use the overloaded method `ChatRoom.publicChat()` directly, which will return `false` for the `PeerChatRoom`.

Reply:
Done!